packages/weather.rb

```diff
@@ -0,0 +1,41 @@
+require 'package'
+
+class Weather < Package
+  description 'This command-line utility is intended to provide quick access to current weather conditions and forecasts.'
+  homepage 'http://fungi.yuggoth.org/weather/'
+  version '2.3'
+  source_url 'http://fungi.yuggoth.org/weather/src/weather-2.3.tar.xz'
+  source_sha256 '86148d2f1d59867f637f52558cc2a6b3280fac94df55c6e5af0ce37cc190d146'
+
+  depends_on 'python27' unless File.exists? "#{CREW_PREFIX}/bin/python"
+
+  def self.build
+    system "echo 'setpath = $HOME/.weather:#{CREW_PREFIX}/data/weather' >> weatherrc"
+  end
+
+  def self.install
+    system "mkdir -p #{CREW_DEST_PREFIX}/bin"
+    system "mkdir -p #{CREW_DEST_PREFIX}/data/weather"
+    system "mkdir -p #{CREW_DEST_PREFIX}/etc"
+    system "mkdir -p #{CREW_DEST_PREFIX}/man/man1"
+    system "mkdir -p #{CREW_DEST_PREFIX}/man/man5"
+    system "cp weather #{CREW_DEST_PREFIX}/bin"
+    system "cp weather.py #{CREW_DEST_PREFIX}/bin"
+    system "cp airports #{CREW_DEST_PREFIX}/data/weather"
+    system "cp places #{CREW_DEST_PREFIX}/data/weather"
+    system "cp slist #{CREW_DEST_PREFIX}/data/weather"
+    system "cp stations #{CREW_DEST_PREFIX}/data/weather"
+    system "cp zctas #{CREW_DEST_PREFIX}/data/weather"
+    system "cp zlist #{CREW_DEST_PREFIX}/data/weather"
+    system "cp zones #{CREW_DEST_PREFIX}/data/weather"
+    system "cp weatherrc #{CREW_DEST_PREFIX}/etc"
+    system "cp weather.1 #{CREW_DEST_PREFIX}/man/man1"
+    system "cp weatherrc.5 #{CREW_DEST_PREFIX}/man/man5"
+  end
+
+  def self.postinstall
+    system "if [ ! -d $HOME/.weather ]; then mkdir $HOME/.weather; fi"
+    system "if [ ! -f $HOME/.weather/weatherrc ]; then cp weatherrc $HOME/.weather; fi"
```

Comment:
maybe
```
cp #{CREW_PREFIX}/etc/weatherrc $HOME/.weather
```
or
```
ln -s #{CREW_PREFIX}/etc/weatherrc $HOME/.weather
```

Reply:
For the reasons you explained yourself earlier, what happens when this package is updated?  Wouldn't we lose changes in `$HOME/.weather/weatherrc` if we did this?

Reply:
Sry, I meant
```
if [ ... ]; then cp #{CREW_PREFIX}/etc/whetherrc $HOME/.weather; fi
```
Or similar to use `ln -s`.  I just wanted to say there is no weatherrc file in the current working directory when install from pre-compiled binary distribution.

Reply:
I'm doing the same exact thing in the install function.  Believe me, this works. :)

Reply:
Unfortunately, no.
```
$ crew install weather
weather: This command-line utility is intended to provide quick access to current weather conditions and forecasts.
http://fungi.yuggoth.org/weather/
version 2.3
Precompiled binary available, downloading...
######################################################################## 100.0%
curl: (33) HTTP server doesn't seem to support byte ranges. Cannot resume.
Archive downloaded
Unpacking archive, this may take a while...
Installing...
Performing post-install...
cp: cannot stat 'weatherrc': No such file or directory
weather failed to install: `if [ ! -f $HOME/.weather/weatherrc ]; then cp weatherrc $HOME/.weather; fi` exited with 1
```
You can test it by copying pre-compiled `tar.xz` file into your `/usr/local/tmp/crew` directory and prepare correct sha256sum and dummy URL.

Reply:
That's not what I get:
```
$ tar xvf weather-2.3.tar.xz 
weather-2.3/
weather-2.3/overrides.conf
weather-2.3/coop-stations.txt
weather-2.3/metar.tbl
weather-2.3/airports
weather-2.3/NEWS
weather-2.3/2015_Gaz_cousubs_national.zip
weather-2.3/INSTALL
weather-2.3/weatherrc.5
weather-2.3/bp16jn16.dbx
weather-2.3/2015_Gaz_zcta_national.zip
weather-2.3/slist
weather-2.3/weather.py
weather-2.3/qa.log
weather-2.3/weatherrc  <<<<< file exists
weather-2.3/zlist
weather-2.3/2015_Gaz_place_national.zip
weather-2.3/zonecatalog.curr.tar
weather-2.3/overrides.log
weather-2.3/places
weather-2.3/FAQ
weather-2.3/zones
weather-2.3/weather
weather-2.3/README
weather-2.3/LICENSE
weather-2.3/weather.1
weather-2.3/stations
weather-2.3/2015_Gaz_counties_national.zip
weather-2.3/zctas
weather-2.3/nsd_cccc.txt
```

Reply:
```
$ crew remove weather
Weather removed!
$ cp weather.rb /usr/local/lib/crew/packages/
$ rm -rf ~/.weather/
$ crew install weather
weather: This command-line utility is intended to provide quick access to current weather conditions and forecasts.
http://fungi.yuggoth.org/weather/
version 2.3
No precompiled binary available for your platform, downloading source...
--2017-08-25 18:35:05--  http://fungi.yuggoth.org/weather/src/weather-2.3.tar.xz
Resolving fungi.yuggoth.org... 162.242.242.62
Connecting to fungi.yuggoth.org|162.242.242.62|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 7056760 (6.7M) [application/x-xz]
Saving to: 'weather-2.3.tar.xz'

weather-2.3.tar.xz                                   100%[====================================================================================================================>]   6.73M  4.98MB/s    in 1.4s    

2017-08-25 18:35:06 (4.98 MB/s) - 'weather-2.3.tar.xz' saved [7056760/7056760]

Archive downloaded
Unpacking archive, this may take a while...
gcc (GCC) 4.9.4
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Building from source, this may take a while...
Preconfiguring package...
Installing...
Performing post-install...
Weather installed!
$ weather "kobe, jp"
Searching via name...
[using result Kobe, JP]
Current conditions at <UNKNOWN>
Last updated Aug 25, 2017 - 07:00 PM EDT / 2017.08.25 2300 UTC
   Temperature: 77 F (25 C)
   Relative Humidity: 73%
   Wind: from the E (080 degrees) at 18 MPH (16 KT)
   Weather: showers in the vicinity
   Sky conditions: mostly cloudy
```

Reply:
What is the output of `echo $HOME`?

Reply:
I'm talking about pre-compiled binary distribution which we are preparing for bintray.  I would like to make this weather package work fine for both source build and pre-compiled binary distribution.  So, I suggested how to do it.  It's easy.  Just adding `#{CREW_PREFIX}/etc/` to one line.  But, you are saying it works believe you like a broken record.  You never heard others.  I'm really tired to talk with you.

Reply:
Oh, you are saying pre-compiled binary.  That means pre (or before) the binary (or source) build.  I misunderstood.  I think you mean just compiled or post-compiled binary.  That makes more sense.  I'll just make the change and not tire you out with my obvious stupidity.